planner/core/rule_join_reorder.go

```diff
@@ -44,7 +44,8 @@ func extractJoinGroup(p LogicalPlan) (group []LogicalPlan, eqEdges []*expression
 	}
 	if !isJoin || join.preferJoinType > uint(0) || join.StraightJoin ||
 		(join.JoinType != InnerJoin && join.JoinType != LeftOuterJoin && join.JoinType != RightOuterJoin) ||
-		((join.JoinType == LeftOuterJoin || join.JoinType == RightOuterJoin) && join.EqualConditions == nil) {
+		// (A * B) left outer join C on (A.a = C.a && B.b = C.b) will generate two left outer join type edge, which is not accepted yet.
```

Comment:
Even for `(A * B) left outer join C on (A.a = C.a && A.b = C.b) `, is it still not allowed?

Reply:
make sense, fixed